tests/playwright/src/model/ai-lab-page.ts

```diff
@@ -0,0 +1,132 @@
+
+  constructor(page: Page, webview: Page) {
+    this.page = page;
+    this.webview = webview;
+    this.heading = webview.getByLabel('Welcome to Podman AI Lab');
+    this.navigationBar = webview.getByLabel('PreferencesNavigation');
+
+    this.recipesCatalogButton = this.navigationBar.getByLabel('Recipes Catalog', { exact: true });
+    this.runningAppsButton = this.navigationBar.getByLabel('Running');
+    this.catalogButton = this.navigationBar.getByLabel('Catalog', { exact: true });
+    this.servicesButton = this.navigationBar.getByLabel('Services');
+    this.playgroundsButton = this.navigationBar.getByLabel('Playgrounds');
+
+    this.recipesCatalogPage = webview.getByRole('region', { name: 'Recipe Catalog' })
+    this.recipesCatalogContent = this.recipesCatalogPage.getByRole('region', { name: 'content', exact: true }).first();
+    this.recipesCatalogNaturalLanguageProcessing = this.recipesCatalogContent.getByRole('region', { name: 'Natural Language Processing', exact: true });
+    this.recipesCatalogAudio = this.recipesCatalogContent.getByRole('region', { name: 'Audio', exact: true });
+    this.recipesCatalogComputerVision = this.recipesCatalogContent.getByRole('region', { name: 'Computer Vision', exact: true });
+
+    this.applicationDetailsPanel = webview.getByLabel('application details panel');
+    this.startAIAppButton = this.applicationDetailsPanel.getByRole('button', { name: 'Start AI App' });
+    this.openAIAppButton = this.applicationDetailsPanel.getByRole('button', { name: 'Open AI App' });
+    this.deleteAIAppButton = this.applicationDetailsPanel.getByRole('button', { name: 'Delete AI App' });
+  }
+
+  async waitForPageLoad(): Promise<void> {
+    await playExpect(this.heading).toBeVisible();
+    await playExpect(this.navigationBar).toBeVisible();
+    await playExpect(this.recipesCatalogButton).toBeVisible();
+    await playExpect(this.runningAppsButton).toBeVisible();
+    await playExpect(this.catalogButton).toBeVisible();
+    await playExpect(this.servicesButton).toBeVisible();
+    await playExpect(this.playgroundsButton).toBeVisible();
+  }
+
+  async openRecipesCatalog(): Promise<void> {
+    await playExpect(this.recipesCatalogButton).toBeVisible();
+    await this.recipesCatalogButton.click();
+    await this.page.waitForTimeout(2000);
+    await playExpect(this.recipesCatalogPage).toBeVisible();
+    await playExpect(this.recipesCatalogContent).toBeVisible();
+    await playExpect(this.recipesCatalogNaturalLanguageProcessing).toBeVisible();
+    await playExpect(this.recipesCatalogAudio).toBeVisible();
+    await playExpect(this.recipesCatalogComputerVision).toBeVisible();
+  }
+
+  async openRecipesCatalogApp(category: Locator, appName: string): Promise<void> {
+    await this.openRecipesCatalog();
+    await category.getByRole('region', { name: 'content', exact: true }).first().getByRole('region', { name: appName, exact: true }).getByRole('button', { name: 'More details' }).click();
```

Comment:
This looks like another class that could be implemented? It is another page I think.

Reply:
created follow-up issue https://github.com/containers/podman-desktop-extension-ai-lab/issues/1476

Reply:
done.